test/tabs/test-firefox-tabs.js

```diff
@@ -1057,6 +1059,41 @@ exports.testFaviconGetterDeprecation = function (assert, done) {
   });
 }
 
+// related to bug 942511
+exports['test active tab properties defined on popup closed'] = function (assert, done) {
+    let defaultOpenWindowPref = getPref(OPEN_IN_NEW_WINDOW_PREF);
+    let defaultPopupPref = getPref(DISABLE_POPUP_PREF);
+
+    setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+    setPref(DISABLE_POPUP_PREF, false);
+
+    let tabID = "";
+
+    tabs.open({
+      url: 'about:blank',
+      onReady: function (tab) {
+        tabID = tab.id;
+        tab.attach({
+          contentScript: 'var popup = window.open("about:blank");' +
+                         'popup.close();'
+        });
+
+        setTimeout(function() {
+          let activeTabID = tabs.activeTab.id;
+          if (activeTabID) {
+              assert.equal(tabID, activeTabID, 'ActiveTab properties are correct');
+          } 
+          else {
+            assert.fail('ActiveTab properties undefined on popup closed');
+          }
+          setPref(OPEN_IN_NEW_WINDOW_PREF, defaultOpenWindowPref);
+          setPref(DISABLE_POPUP_PREF, defaultPopupPref);
```

Comment:
we should add before and after tests to this file to make sure that these prefs are in the state they should be.

For an example of using these see the [sidebar tests](https://github.com/colbat/addon-sdk/blob/activetab-properties-undefined-942511/test/test-ui-sidebar.js#L1545-L1552)